### DISCUSSION TITLE
Make WebAuthn required actions to be registered in the realms by default

### DISCUSSION THREAD

**Discussion by mposolda** [keycloak/keycloak#9523] [Ideas]
Currently when WebAuthn is used, administrator needs to manually register WebAuthn related required actions `WebAuthn Register` and `WebAuthn Register Passwordless`. I think that once WebAuthn become supported feature, it will makes sense to register WebAuthn related required actions `WebAuthn Register` and `WebAuthn Register Passwordless` in the new realm by default. This will be compatible with other required actions (EG. OTP registration) and it will simplify the WebAuthn setup for administrators.

WDYT?

**Comment by mposolda** [comment]
Other option is, that we can improve the WebAuthn integration experience with authentication policies. The ideal experience is, that after single click "Enable WebAuthn as 2nd  factor" will take care of register WebAuthn required action as well as using WebAuthn as authenticator.

The design proposal for Authentication policies should come soon.

**Reply by tnorimat** [reply]
@mposolda Could you elaborate the concept "Authentication Policies"?
